velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -235,23 +402,77 @@ void WindowFuzzer::go() {
     const auto call =
         makeFunctionCall(signature.name, argNames, false, false, ignoreNulls);
 
+    auto [frameType, startBoundType, endBoundType] =
+        frameWindowTypeAndBoundType();
+
+    std::vector<core::WindowNode::BoundType> kBoundOptions = {
+        core::WindowNode::BoundType::kPreceding,
+        core::WindowNode::BoundType::kFollowing};
+    bool isKRangeFrame = frameType == core::WindowNode::WindowType::kRange &&
+        (std::find(
+             kBoundOptions.begin(), kBoundOptions.end(), startBoundType) !=
+             kBoundOptions.end() ||
+         std::find(kBoundOptions.begin(), kBoundOptions.end(), endBoundType) !=
+             kBoundOptions.end());
+
     std::vector<SortingKeyAndOrder> sortingKeysAndOrders;
+    TypeKind orderByTypeKind;
+    std::string orderByKey = "";
     // 50% chance without order-by clause.
-    if (vectorFuzzer_.coinToss(0.5)) {
+    if (isKRangeFrame) {
+      // k-range frames need 1 order by key
+      sortingKeysAndOrders =
+          generateSortingKeysAndOrders("s", argNames, argTypes, isKRangeFrame);
+      orderByTypeKind = argTypes.back()->kind();
+      orderByKey = sortingKeysAndOrders[0].key_;
+    } else if (vectorFuzzer_.coinToss(0.5)) {
       sortingKeysAndOrders =
           generateSortingKeysAndOrders("s", argNames, argTypes);
     }
     const auto partitionKeys = generateSortingKeys("p", argNames, argTypes);
-    const auto [frameClause, isRowsFrame] = generateFrameClause();
-    const auto input = generateInputDataWithRowNumber(
-        argNames, argTypes, partitionKeys, signature);
+
     // If the function is order-dependent or uses "rows" frame, sort all input
     // rows by row_number additionally.
-    if (requireSortedInput || isRowsFrame) {
+    if ((requireSortedInput && !isKRangeFrame) ||
+        frameType == core::WindowNode::WindowType::kRows) {
       sortingKeysAndOrders.emplace_back("row_number", core::kAscNullsLast);
       ++stats_.numSortedInputs;
     }
 
+    auto input = generateInputDataWithRowNumber(
+        argNames, argTypes, partitionKeys, signature, orderByKey);
+    if (isKRangeFrame) {
+      try {
+        VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
+            addOffsetColumnsToInput,
+            orderByTypeKind,
+            input,
+            startBoundType,
+            endBoundType,
+            sortingKeysAndOrders[0]);
+      } catch (VeloxUserError& e) {
+        // NOTE: velox user exception is accepted as it is caused by the invalid
+        // fuzzer test inputs. Catches arithmetic overflow exceptions in case
+        // value of orderBy column +/- offset exceeds the datatype limits.
+        stats_.numFailed++;
```

Comment:
No need to increment numFailed since the query doesn't run at all. This counter is for queries that run and fail.

Reply:
Reverted, thanks for pointing it.